todo_app/data/Trello.py

```diff
@@ -0,0 +1,119 @@
+from todo_app.data.ResponseFilter import ResponseFilter
+from todo_app.ListUtils import swap_key_in_list, combine_lists
+import requests
+import os
+
+
+class trello:
+    api_key = os.environ.get('API_KEY')
+    api_token = os.environ.get('API_TOKEN')
+    trello_board_id = os.environ.get('BOARD_ID')
+
+    todo_name = 'To Do'
+    in_progress = 'Doing'
+    done = 'Done'
+    todo_id = None
+    in_progress_id = None
+    done_id = None
+
+    base_url = 'https://api.trello.com/1/'
+
+    headers = {'key': api_key, 'token': api_token}
+
+    fields_from_cards = {'fields': f'{ResponseFilter.name},{ResponseFilter.id_list}'}
+
+    get_cards_header = headers | fields_from_cards
+
+    get_all_cards_url = f'{base_url}boards/{trello_board_id}/cards'
+
+    fields_from_list = {ResponseFilter.fields: f'{ResponseFilter.name},idBoard'}
+
+    get_lists_headers = headers | fields_from_list
+
+    get_lists_url = f'{base_url}boards/{trello_board_id}/lists'
+    cards_url = f'{base_url}cards'
+
+    lists_from_board = None
+
+    @classmethod
+    def get_lists(cls):
+        if cls.lists_from_board is None:
+            lists = requests.get(
+                cls.get_lists_url, params=cls.get_cards_header).json()
+            lists = swap_key_in_list(
+                lists, ResponseFilter.id, ResponseFilter.id_list)
+            cls.lists_from_board = swap_key_in_list(
+                lists, ResponseFilter.name, ResponseFilter.status)
+
+        return cls.lists_from_board
+
+    @classmethod
+    def get_all_list_ids(cls, list_name):
+        lists = cls.get_lists()
+        return next((item for item in lists if item[ResponseFilter.status] == list_name), None)[ResponseFilter.id_list]
+
+    @classmethod
+    def get_todo_list_id(cls):
+        if cls.todo_id is None:
+            cls.todo_id = cls.get_all_list_ids(cls.todo_name)
+
+        return cls.todo_id
+
+    @classmethod
+    def get_doing_list_id(cls):
+        if cls.in_progress_id is None:
+            cls.in_progress_id = cls.get_all_list_ids(cls.in_progress)
+
+        return cls.in_progress_id
+
+    @classmethod
+    def get_done_list_id(cls):
+        if cls.done_id is None:
+            cls.done_id = cls.get_all_list_ids(cls.done)
+
+        return cls.done_id
+
+    @staticmethod
+    def get_list_items():
+        cards = requests.get(
+            trello.get_all_cards_url, params=trello.get_cards_header).json()
+        lists = trello.get_lists()
+        items = combine_lists(
+            cards, lists, ResponseFilter.id_list)
```

Comment:
You could get rid of the `get_all_items` and simplify `get_lists` if you instead do something like the following. (And then by extension you could also delete the `combine_lists` and `swap_key_in_list` functions. And I would say you can delete the Items class.)

```
items = []
for card in cards:
    status = search_list(lists, 'id', card['idList'])['name']
    items.append(Trello_Item(card['id'], card['name'], status))
```

You could still extract the logic for getting a list name into a function. And you could of course keep using your ResponseFilter variables instead of raw strings.

Your current solution works but:
- It requires massaging dictionaries into the right shape, making this line implicitly dependent on stuff happening inside `get_lists`.
- It's nice to move from dictionaries to the TrelloItem class sooner rather than later. With my suggestion, this is the only function in the codebase that will care about the structure of the card data you are receiving from Trello.

As a side benefit, this also reduces the amount of code you need.